docs/src/userguide/config/event_config.rst

```diff
@@ -299,10 +299,100 @@ Example configuration for `Google BigQuery`_, by specifying a query in the BigQu
 
     Inherited from :ref:`global event input config <skip-klio-read>`.
 
+
+.. _event-config-avro-read:
+
+Avro
+^^^^
+
+*Mode: Batch*
+
+``KlioReadFromAvro`` reads records in from the provided avro location.
+If the records' schema include an ``element`` field,
+the KlioMessage ``data.element`` will be set to the value.
+If there is no ``element`` field on the records read in from the avro,
+the entire record will be cast to bytes and stuffed into
+the KlioMessage ``data.element`` field.
+
+Example configuration for reading elements from a directory of avro files:
+
+.. code-block:: yaml
+
+    name: my-cool-batch-job
+    pipeline_options:
+        streaming: False
+    job_config:
+        events:
+            inputs:
+                - type: avro
+                  location: gs://my-bucket/
+
+Example configuration for reading elements from a specific avro file:
+
+.. code-block:: yaml
+
+    name: my-cool-batch-job
+    pipeline_options:
+        streaming: False
+    job_config:
+        events:
+            inputs:
+                - type: avro
+                  location: gs://my-bucket
+                  file_pattern: my_specific_file.avro
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``avro``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].location STR
+
+    Location of local or GCS avro file(s) to read.
+    This field is optional if ``job_config.events.inputs[].file_pattern`` is provided.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].file_pattern STR
+
+    Pattern of file name(s) to read.
+    This field is optional if ``job_config.events.inputs[].location`` is provided.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+    .. note::
+
+        If both ``job_config.events.inputs[].location`` and
+        ``job_config.events.inputs[].file_pattern`` are provided,
+        the two fields will be joined to to find files matching ``file_pattern``
+        located in the provided ``location`` path.
+
+.. option:: job_config.events.inputs[].min_bundle_size INT
+
+    Minimum size (bytes) considered when splitting the input into bundles.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].validate BOOL
+
+    flag to verify files exist during pipeline creation time.
```

Comment:
s/flag/Flag

Also, it'd probably be good to add something like "Defaults to `True`."